### PR TITLE
Pass more strings using as string_views

### PR DESCRIPTION
**Pull request by Broxzier**
As well as changing `std::string&` to `u8string_view` in some places where appropriate, like for paths.

### CODE REVIEW

File: src/openrct2/core/FileScanner.cpp [hunk 1]
```diff
@@ -277,15 +277,15 @@ class FileScannerUnix final : public FileScannerBase
     void GetDirectoryChildren(std::vector<DirectoryChild>& children, std::string_view path) override
     {
         struct dirent** namelist;
-        int32_t count = scandir(path.c_str(), &namelist, FilterFunc, alphasort);
+        int32_t count = scandir(path.data(), &namelist, FilterFunc, alphasort);
```
Comment (AaronVanGeffen) [inline src/openrct2/core/FileScanner.cpp:280]: Is this guaranteed to be null-terminated?

Comment (Broxzier) [reply src/openrct2/core/FileScanner.cpp:280]: Short answer: It _is_ possible that the actual data isn't null-terminated, but you'd have to do your best to get there.

The `string_view::data` member function itself doesn't guarantee that its data is null-terminated, however `path` gets implicitly constructed from null-terminated strings, which under the hood uses `strlen` to determine the length. You'd have to manually provide a length, take a substring of a path, or do other shenanigans to make the length it knows different from the actual string.

Comment (duncanspumpkin) [reply src/openrct2/core/FileScanner.cpp:280]: This also has the added bug of letting you read beyond the size of the path.
```
std::string thing("pathtoitem/ignorethisbit/")
std::string_view thingView = thing;
std::string_view subThingView = thingView.substr(0, 11); // Get just the "pathtoitem/"
GetDirectoryChildren(output, subThingView); // Error scans "pathtoitem/ignorethisbit/"
```

**Review by duncanspumpkin — CHANGES_REQUESTED** [review]
We can't use string_view for a good number of these due to requiring to access the `c_str()` string_view is not guaranteed to be null terminated and it is trivial to make things none null terminated.

**Comment by IntelOrca** [comment]
Anywhere that uses `.data()` for something that takes a `char*` is going to be a problem. You can't safely use `.data()` in any of the C APIs like s/printf.

File: src/openrct2/core/FileScanner.cpp (hunk 1 above)
Comment (Broxzier) [reply src/openrct2/core/FileScanner.cpp:280]: Yeah, that's what I was saying, an unlikely scenario. But I agree that it's better to avoid this at all.

**Comment by CookiePLMonster** [comment]
For C API's, consider rolling out a specialization of `std::string_view` like [`wil::zstring_view`](https://github.com/microsoft/wil/blob/5eb59d60e167482639cc47ffb40442158da7fd04/include/wil/stl.h#L121-L127) that retains string view remantics while guaranteeing null termination.

**Comment by Broxzier** [comment]
Closing this, to split it into smaller parts. I initially started with InteractiveConsole, so I'll just start from there again.

**Pull request closed without merging**